### PR TITLE
[upd] Add a zero-volume indication

### PR DESCRIPTION
**Pull request by YanDoroshenko**
Not sure this can be merged as is.
audio-volume-zero-symbolic.svg is not included in every icon theme and if it is absent, a placeholder for a missing icon (like a broken image frame) is displayed. Maybe it should be enabled with a feature flag or something?

### CODE REVIEW

**Comment by christophgysin** [comment]
> audio-volume-zero-symbolic.svg is not included in every icon theme

Is there a way to check and dynamically fall back to another icon? If so, I'd be happy to merge this.

**Comment by YanDoroshenko** [comment]
How does this look? I've restructured fallbacks so that:
* a volume level uses the next one as a fallback
* a mic level uses the next mic level as a fallback and if it's out of mic levels, it tries to use the relevant volume level (with volume level fallbacks)

**Comment by YanDoroshenko** [comment]
@christophgysin What do you think?

File: configure.ac [hunk 1]
```diff
@@ -19,7 +19,7 @@
 
 AC_PREREQ([2.68])
 
-AC_INIT([pasystray], [0.8.0], [christoph.gysin@gmail.com], [pasystray],
+AC_INIT([pasystray], [0.9.0], [christoph.gysin@gmail.com], [pasystray],
```
Comment (christophgysin) [inline configure.ac:22]: Don't bump the version.

File: src/ui.c [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (christophgysin) [inline src/ui.c:1]: Don't add empty line.

File: src/ui.c [hunk 3]
```diff
@@ -150,27 +156,38 @@ GtkDialog* ui_errordialog(const gchar* title, const gchar* message)
...
+                    ui_icon_names[mt][i] = ui_find_icon_name(theme, volume_icon_names, i, settings->symbolic_icons);
+                }
+                g_debug(g_strdup_printf("Using microphone icon %s", ui_icon_names[mt][i]));
+            }
+        }
+        else {
+            for(icon_idx_t i = 0; i < ICON_IDX_COUNT; i++) {
+                ui_icon_names[mt][i] = ui_find_icon_name(theme, volume_icon_names, i, settings->symbolic_icons);
+                g_debug(g_strdup_printf("Using speaker icon %s", ui_icon_names[mt][i]));
+            }
```
Comment (christophgysin) [inline src/ui.c:190]: I'm not quite happy with this. There too much special cases and duplication. Also, I don't think it's necessary to have falback for all icons, only for symbolic ones and the *-level-zero ones, which might not be available yet for all users.

Therefore I would suggest to refactor the icon_set_t into a structure that accepts a list of icon, e.g.:
```c
static icon_set_t volume_icon_names = {
    [ICON_IDX_MUTED] = ["audio-volume-muted", NULL],
    [ICON_IDX_ZERO] = ["audio-volume-zero", "audio-volume-muted", NULL],
    [ICON_IDX_LOW] = ["audio-volume-low", NULL],
```
That should then make the implementation a lot more generic again.

**Review by christophgysin — CHANGES_REQUESTED** [review]

**Pull request closed without merging**